### PR TITLE
Mongolid Migration v4 - Part 1

### PR DESCRIPTION
**Pull request by djonasm**
Get some features from mongolid 3 to start a brand new version with baby steps modifications

### CODE REVIEW

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...t #8":0,"Mongolid\\Util\\ObjectIdUtilsTest::testShouldEvaluateIfValueIsAnObjectid with data set #9":0,"Mongolid\\Util\\ObjectIdUtilsTest::testShouldEvaluateIfValueIsAnObjectid with data set #10":0,"Mongolid\\Util\\ObjectIdUtilsTest::testShouldEvaluateIfValueIsAnObjectid with data set #11":0,"Mongolid\\Util\\ObjectIdUtilsTest::testShouldEvaluateIfValueIsAnObjectid with data set #12":0,"Mongolid\\Util\\ObjectIdUtilsTest::testShouldEvaluateIfValueIsAnObjectid with data set #13":0,"Mongolid\\Util\\SequenceServiceTest::testShouldGetNextValue":0.002,"Mongolid\\Util\\SequenceServiceTest::testShouldGetNextValue with data set \"Returned as an array instead of object\"":0.002,"Mongolid\\Util\\SequenceServiceTest::testShouldGetRawCollection":0.002,"Mongolid\\Connection\\ConnectionTest::testShouldGetConnection":0,"Mongolid\\Connection\\ManagerTest::testShouldAddAndGetConnection":0.031,"Mongolid\\Connection\\ManagerTest::testShouldSetEventTrigger":0.004,"Mongolid\\Connection\\ManagerTest::testShouldInitializeOnce":0.001,"Mongolid\\Container\\ContainerTest::testShouldCallMethodsProperlyWithNoArguments":0.019,"Mongolid\\Container\\ContainerTest::testShouldCallMethodsProperlyWithArguments":0.001}}
```
Comment (diegofelix) [inline .phpunit.result.cache:1]: Remove this file and put it on .gitignore.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -9,19 +9,22 @@
          convertWarningsToExceptions="true"
          processIsolation="false"
          stopOnFailure="false"
-         syntaxCheck="false"
 >
     <testsuites>
-        <testsuite name="Package Test Suite">
-            <directory>./tests/</directory>
+        <testsuite name="unit">
```
Comment (diegofelix) [inline phpunit.xml.dist:14]:
```suggestion
        <testsuite name="Package Unit Tests">
```

File: src/Event/EventTriggerService.php [hunk 3]
```diff
@@ -45,4 +45,9 @@ public function fire(string $event, $payload, bool $halt = false)
 
         return true;
     }
+
+    public function getDispatcher():EventTriggerInterface
```
Comment (diegofelix) [inline src/Event/EventTriggerService.php:49]:
```suggestion
    public function getDispatcher(): EventTriggerInterface
```

File: tests/Mongolid/Connection/PoolTest.php [hunk 4]
```diff
@@ -1,72 +0,0 @@
-<?php
```
Comment (diegofelix) [inline tests/Mongolid/Connection/PoolTest.php:1]: Did you deleted this for mistake?

File: tests/Unit/Container/ContainerTest.php [hunk 5]
```diff
@@ -4,11 +4,11 @@
 
 use Illuminate\Container\Container;
 use Mockery as m;
-use TestCase;
+use Mongolid\TestCase;
 
 class IocTest extends TestCase
 {
-    public function tearDown()
+    public function setUp(): void
```
Comment (diegofelix) [inline tests/Unit/Container/ContainerTest.php:11]: I don't know how it is working....

```suggestion
    public function tearDown(): void
```

**Review by diegofelix — CHANGES_REQUESTED** [review]

File: tests/Unit/ActiveRecordTest.php [hunk 6]
```diff
@@ -33,28 +34,13 @@ public function setUp()
...
-                'created_at' => 'createdAtTimestamp',
-                'updated_at' => 'updatedAtTimestamp',
-            ],
-            'fields',
-            $this->entity
-        );
-        $this->assertTrue($this->entity->dynamic);
-    }
-
     public function testShouldImplementModelTraits()
```
Comment (ezandonai) [inline tests/Unit/ActiveRecordTest.php:44]:
```suggestion
    public function testShouldImplementModelTraits(): void
```

File: tests/Unit/ActiveRecordTest.php [hunk 7]
```diff
@@ -372,42 +358,39 @@ public function testShouldGetDataMapper()
         $this->assertEquals($schema, $result->getSchema());
     }
 
-    /**
-     * @expectedException \Mongolid\Exception\NoCollectionNameException
-     */
     public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallAllFunction()
```
Comment (ezandonai) [inline tests/Unit/ActiveRecordTest.php:361]:
```suggestion
    public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallAllFunction(): void
```

File: tests/Unit/ActiveRecordTest.php [hunk 8]
```diff
@@ -372,42 +358,39 @@ public function testShouldGetDataMapper()
...
 
+        $this->expectException(NoCollectionNameException::class);
+
         $entity->all();
     }
 
-    /**
-     * @expectedException \Mongolid\Exception\NoCollectionNameException
-     */
     public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallFirstFunction()
```
Comment (ezandonai) [inline tests/Unit/ActiveRecordTest.php:373]:
```suggestion
    public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallFirstFunction(): void
```

File: tests/Unit/ActiveRecordTest.php [hunk 9]
```diff
@@ -372,42 +358,39 @@ public function testShouldGetDataMapper()
...
 
+        $this->expectException(NoCollectionNameException::class);
+
         $entity->first();
     }
 
-    /**
-     * @expectedException \Mongolid\Exception\NoCollectionNameException
-     */
     public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallWhereFunction()
```
Comment (ezandonai) [inline tests/Unit/ActiveRecordTest.php:385]:
```suggestion
    public function testShouldRaiseExceptionWhenHasNoCollectionAndTryToCallWhereFunction(): void
```

**Review by ezandonai — APPROVED** [review]

**Pull request merged**